ADR/0033-product-version-controller.md

```diff
@@ -0,0 +1,252 @@
+      would be `v1`.
+
+* The new *ProjectDevStreamTemplate* will include a *version* variable with no
+  default value.
+
+* All *Application* and *Component* resources that are associated with the
+  cloned *ProjectDevStream* will be copied into the *resources* section of the
+  *ProjectDevStreamTemplate* resource, while *status* and other transient
+  properties would be stripped.
+
+* Resources names within the template will be set as following:
+
+    * If the name on the cloned resource ends with the *ProjectDevStream*
+      resource's version value as determined above, the value is stripped and
+      the "`{version}`" template string is places instead.
+    * Otherwise the "`{version}`" template string is appended to the name.
+
+* The template will include variables for customizing *revision* (branch),
+  *dockerfileUrl* and *context* values for all included *Component* resources.
+  The default values for the variables will be the existing values from the
+  cloned objects. For *revision* properties the default values will be
+  determined using the *version* variable in a similar manner to the way
+  resource names are determined.
+
+The generated template ends up being able to generate a *ProjectDevStream* that
+is very similar to the one its being generated from but with many variables that
+would allow customizing many aspects such as Git branches being used.
+
+It is intended that a UI screen could be displayed based on the generated
+template. It is likely that such screen would need to group variables by the
+resource they customize. Variable names will be picked to allow doing so. In
+addition, the screen may need to display human friendly descriptions for the
+variables or refer to the original resources that the created resources are
+cloned from. The "description" property will be added to variables for the
+former while a "pvc.konflux.dev/cloned-from" annotation on the template resource
+sections will allow for the latter.
+
+The process for cloning a *ProjectDevStream* from the UI will be as follows:
+
+1. Let the user select the *ProjectDevStream* resource to be cloned, set the
+   `toTemplate` property on it.
+2. Wait for the *ProjectDevStreamTemplate* resource to be created
+3. Remove the `toTemplate` property to prevent further updates to the template
+   resource by the controller.
+4. Display a screen allowing the user to select a name fo the new
+   *ProjectDevStream*  resource and values for all the template variables. When
+   default values  exist, the screen should display then and let the user keep
+   them.
+5. Once the user confirms the screen, create a *ProjectDevStream* resource with
+   a 'template' property referring to the template and values for the variables
+   as selected by the user.
+6. Once all the child resources specified by the template are created (The state
+   field on the *ProjectDevStream* resource should indicate that) Delete the
+   template.
```

Comment:
Is this the step that prevents further alignment of the applications and components with the projectdevstream's template?

Reply:
That, and line #3 above are in place to prevent all kinds of strange loops.